### PR TITLE
Terraform housekeeping

### PR DESCRIPTION
**Pull request by Stretch96**
* Adds a terraform-docs DotFile
* Removes what appears to be old commented code
* Collates `locals` into single files
* Pins module versions to ensure resources aren't unintentionally modified
* Makes use of `jsonencode` rather than JSON strings for easier readability
* Ran a `terraform fmt` to format the HCL

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -13,15 +12,15 @@ terraform {
...
     key    = "ds-infrastructure-enrichment-pipeline/backend.tfstate"
     region = "eu-west-2"
   }
 }
 
 module "lambda_s3" {
   source = "./modules/lambda_s3"
-  name   = "tna"
+
+  name = "tna"
```
Comment (dragon-dxw) [inline terraform/main.tf:23]: There's a part of me that doesn't like the name 'tna' as not being descriptive, but it's an old name that's not worth changing.

File: terraform/modules/data/locals.tf [hunk 2]
```diff
@@ -7,4 +11,9 @@ locals {
       deletion_protection = true
     }
   }
+
+  tags = {
+    Environment = var.environment
+    Project     = "TNA judgement enrichment"
```
Comment (dragon-dxw) [inline terraform/modules/data/locals.tf:17]: As a super-nitpicky comment: we use "judgment" (no central e) throughout whenever referring to a legal document as a noun.

Comment (dragon-dxw) [reply terraform/modules/data/locals.tf:17]: I'd also generally refer to the project as "Find Case Law" so it's distinct from "the Transformation Engine" etc.

File: terraform/modules/lambda_s3/queue.tf [hunk 3]
```diff
@@ -65,7 +60,10 @@ resource "aws_sqs_queue" "replacements-queue" {
   message_retention_seconds  = 86400
   receive_wait_time_seconds  = 10
   sqs_managed_sse_enabled    = true
-  redrive_policy             = "{\"deadLetterTargetArn\":\"${aws_sqs_queue.replacements_dlq_queue.arn}\",\"maxReceiveCount\":4}"
+  redrive_policy = jsonencode({
```
Comment (dragon-dxw) [inline terraform/modules/lambda_s3/queue.tf:63]: <3

File: terraform/modules/network/local.tf [hunk 4]
```diff
@@ -1,18 +0,0 @@
-locals {
```
Comment (dragon-dxw) [inline terraform/modules/network/local.tf:1]: Curious as to why this isn't needed.

File: terraform/main.tf (hunk 1 above)
Comment (Stretch96) [reply terraform/main.tf:23]: Yep if we change that, it'll delete/recreate a lot of the resources

File: terraform/modules/data/locals.tf (hunk 2 above)
Comment (Stretch96) [reply terraform/modules/data/locals.tf:17]: I'll need to check that this tag isn't being used as a filter (eg. for billing) before we can change this - I'll add a Trello card 👍

**Pull request merged**